internal/storage/bucket_handle.go

```diff
@@ -70,16 +70,34 @@ func (bh *bucketHandle) NewReader(
 func (b *bucketHandle) DeleteObject(ctx context.Context, req *gcs.DeleteObjectRequest) error {
 	obj := b.bucket.Object(req.Name)
 
-	// Switching to the requested generation of the object.
-	if req.Generation != 0 {
-		obj = obj.Generation(req.Generation)
-	}
+	// Switching to the requested generation of the object. By default, generation
+	// is 0 which signifies the latest generation.
```

Comment:
Add a comment that even if we dont set generation it will delete live object. With explicitly setting generation=0 we are satisfying idempotency condition.

Reply:
added.

Reply:
Add one more line saying explicitly setting generation as zero to satisfy idempotency condition.